pkg/scheduler/plugins/deviceshare/device_plugin_adapter.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deviceshare
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/clock"
+
+	apiext "github.com/koordinator-sh/koordinator/apis/extension"
+	schedulingv1alpha1 "github.com/koordinator-sh/koordinator/apis/scheduling/v1alpha1"
+	"github.com/koordinator-sh/koordinator/pkg/scheduler/frameworkext"
+)
+
+const (
+	AnnotationBindTimestamp = apiext.SchedulingDomainPrefix + "/bind-timestamp"
+
+	AnnotationPredicateTime = "predicate-time"
+	AnnotationHuaweiNPUCore = "huawei.com/npu-core"
+)
+
```

Comment:
Here we may need more comments to explain why we need to bind these annotations and what the coordination mechanism is.